### PR TITLE
refactor(ndt_scan_matcher): fixed as pointed out by clang-tidy

### PR DESCRIPTION
**Pull request by SakodaShintaro**
## Description

I modified 
* ndt_scan_matcher_core.hpp
* ndt_scan_matcher_core.cpp

according to the warnings from clang-tidy.

## Tests performed

I have checked the operation of the following three

* logging_simulator
* driving_log_replayer
* awsim

## Effects on system behavior

There are no changes about behavior.

## Pre-review checklist for the PR author

The PR author **must** check the checkboxes below when creating the PR.

- [x] I've confirmed the [contribution guidelines].
- [x] The PR follows the [pull request guidelines].

## In-review checklist for the PR reviewers

The PR reviewers **must** check the checkboxes below before approval.

- [ ] The PR follows the [pull request guidelines].

## Post-review checklist for the PR author

The PR author **must** check the checkboxes below before merging.

- [ ] There are no open discussions or they are tracked via tickets.

After all checkboxes are checked, anyone who has write access can merge the PR.

[contribution guidelines]: https://autowarefoundation.github.io/autoware-documentation/main/contributing/
[pull request guidelines]: https://autowarefoundation.github.io/autoware-documentation/main/contributing/pull-request-guidelines/

### CODE REVIEW

**Comment by SakodaShintaro** [comment]
The clang-tidy check is still in progress, but it seems that another lint-check is causing an error due to the effect of "NOLINT" that was added to avoid clang-tidy warnings.
https://results.pre-commit.ci/run/github/429651616/1689221255.qABtU78hRfiNym2rIKcIUQ

I didn't realize there were two types of lint-check running. I'll fix that later.

**Comment by kminoda** [comment]
Thank you! Let us know if the required CIs are fixed so that we can start reviewing :+1: (Maybe better to make it a draft until then)

File: localization/ndt_scan_matcher/include/ndt_scan_matcher/ndt_scan_matcher_core.hpp [hunk 1]
```diff
@@ -107,18 +108,18 @@ class NDTScanMatcher : public rclcpp::Node
     const bool is_converged);
   void publish_point_cloud(
     const rclcpp::Time & sensor_ros_time, const std::string & frame_id,
-    const pcl::shared_ptr<pcl::PointCloud<PointSource>> & sensor_points_mapTF_ptr);
+    const pcl::shared_ptr<pcl::PointCloud<PointSource>> & sensor_points_in_map_ptr);
   void publish_marker(
     const rclcpp::Time & sensor_ros_time, const std::vector<geometry_msgs::msg::Pose> & pose_array);
   void publish_initial_to_result_distances(
-    const rclcpp::Time & sensor_ros_msg, const geometry_msgs::msg::Pose & result_pose_msg,
+    const rclcpp::Time & sensor_ros_time, const geometry_msgs::msg::Pose & result_pose_msg,
```
Comment (kminoda) [inline localization/ndt_scan_matcher/include/ndt_scan_matcher/ndt_scan_matcher_core.hpp:115]: Thank you so much for finding this...!

File: localization/ndt_scan_matcher/src/ndt_scan_matcher_core.cpp [hunk 2]
```diff
@@ -696,22 +698,21 @@ void NDTScanMatcher::service_trigger_node(
...
   // get TF from pose_frame to map_frame
-  auto TF_pose_to_map_ptr = std::make_shared<geometry_msgs::msg::TransformStamped>();
+  auto tf_pose_to_map_ptr = std::make_shared<geometry_msgs::msg::TransformStamped>();
   tf2_listener_module_->get_transform(
-    get_clock()->now(), map_frame_, req->pose_with_covariance.header.frame_id, TF_pose_to_map_ptr);
+    get_clock()->now(), map_frame_, req->pose_with_covariance.header.frame_id, tf_pose_to_map_ptr);
 
   // transform pose_frame to map_frame
-  const auto mapTF_initial_pose_msg = transform(req->pose_with_covariance, *TF_pose_to_map_ptr);
+  const auto initial_pose_in_map_frame = transform(req->pose_with_covariance, *tf_pose_to_map_ptr);
```
Comment (kminoda) [inline localization/ndt_scan_matcher/src/ndt_scan_matcher_core.cpp:713]: Why not this? (Was it too long?)
```suggestion
  const auto initial_pose_msg_in_map_frame = transform(req->pose_with_covariance, *tf_pose_to_map_ptr);
```

**Review by kminoda — COMMENTED** [review]
Thank you very much for the PR :pray:

**Comment by KYabuuchi** [comment]
I found that spell-check failed due to `undistortion`, so I have submitted a PR to add it into the dictionaly. https://github.com/tier4/autoware-spell-check-dict/pull/563

image

File: localization/ndt_scan_matcher/src/ndt_scan_matcher_core.cpp (hunk 2 above)
Comment (SakodaShintaro) [reply localization/ndt_scan_matcher/src/ndt_scan_matcher_core.cpp:713]: I'm a bit concerned about the long variable names, but I certainly think your suggestion is better. I will fix it later.

Comment (SakodaShintaro) [reply localization/ndt_scan_matcher/src/ndt_scan_matcher_core.cpp]: I have done at below commit.
https://github.com/autowarefoundation/autoware.universe/pull/4270/commits/3e018bc9cec1cd028f3965a03d9ac625d1b9716a

**Review by kminoda — APPROVED** [review]
LGTM! Thank you for the PR :pray:

**Pull request merged**